src/main/java/com/cloudbees/jenkins/support/impl/SlaveCommandStatistics.java

```diff
@@ -84,6 +100,16 @@ public boolean shouldBeFiltered() {
         }));
     }
 
+    @VisibleForTesting
+    /*package*/ Map<String, Statistics> getStatistics() {
+        synchronized (statLock) {
+            Map<String, Statistics> out = new HashMap<>(statistics.size() + statLog.size());
```

Comment:
This discards the sort order. Recommend a `TreeMap` here (in which case `statistics` would be a `HashMap`).

Reply:
Agreed with preserving the sort order. Do not understand the "in which case statistics would be a HashMap" path.

Reply:
Or a `LinkedHashMap`, preserving sorted `statistics` followed by chronological `statLog`. Whatever.